indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2113,7 +2113,13 @@ public Boolean apply(Pair<SeekableStreamIndexTaskRunner.Status, Map<PartitionIdT
     for (int i = 0; i < results.size(); i++) {
       String taskId = futureTaskIds.get(i);
       if (results.get(i).isError() || results.get(i).valueOrThrow() == null) {
-        killTask(taskId, "Task [%s] failed to return status, killing task", taskId);
+        log.info("Task [%s] failed to respond, might need to kill it.", taskId);
+        Optional<TaskStatus> killTaskStatus = taskStorage.getStatus(taskId);
+        if (killTaskStatus.isPresent() && !killTaskStatus.get().isComplete()) {
+          // If the task completed while we were failing to chat with it, don't do anything.
+          log.info("Task [%s] failed to return status, killing task", taskId);
```

Comment:
This log may be redundant as in `TaskRunner#shutdown` the reason has been also logged.

Reply:
i think technically thats a implementation detail since TaskRunner is a interface